internal/worker/externalcontrollerupdater/externalcontrollerupdater.go

```diff
@@ -243,6 +244,12 @@ func (w *controllerWatcher) loop() error {
 			if err == w.catacomb.ErrDying() {
 				return err
 			} else if err != nil {
+				// If the controller cannot report its own addresses e.g. if it's
+				// behind a load-balancer then stop the worker without error.
+				if params.IsCodeNotSupported(err) {
+					logger.Debugf("assuming controller cannot report its own addresses and completing watch for %q", w.tag.Id())
```

Comment:
nit: should we still log this as an error instead of debug?

Reply:
I think error is probably too high. Warning seems okay, as does debug imo.